WalletWasabi/Blockchain/BlockFilters/IndexBuilderService.cs

```diff
@@ -319,24 +240,30 @@ public void Synchronize()
 			});
 		}
 
-		private async Task<SyncInfo> GetSyncInfoAsync()
+		private HashSet<Script> FetchScripts(VerboseBlockInfo block)
 		{
-			var bcinfo = await RpcClient.GetBlockchainInfoAsync();
-			var pbcinfo = new SyncInfo(bcinfo);
-			return pbcinfo;
-		}
+			var scripts = new HashSet<Script>();
```

Comment:
Yes, the usage of a `HashSet` here seems to make a lot of sense but it is really unnecessary because the `GolombRiceFilterBuilder` already uses a `HashSet` internally to make sure every entry is unique and there are no duplications so, we are doing more work without any benefit.

In fact, this method makes sense only because it uses a `HashSet`because otherwise it could be replaces by a linq query like:

```c#
private IEnumerable<Script> FetchScripts(VerboseBlockInfo block)
{
	return block.Transactions.SelectMany( tx =>
			tx.Inputs
			.Select(i => i.PrevOutput)
			.Concat(tx.Outputs)
			.Where(o => o is { } && o.PubkeyType == RpcPubkeyType.TxWitnessV0Keyhash)
			.Select(o => o.ScriptPubKey));
}
```

Reply:
I am now generally less keener on long linq queries I don't think the complexity justifies the readability.

I can defo remove the hashset though good spot.

Reply:
FYI, I don't mind using this linq query if you think its more readable I will use that.

Reply:
It is fine line this.